browser-test/src/reporters/file_placement_reporter.ts

```diff
@@ -0,0 +1,158 @@
+   * @param specs An array of elements from a specs property from the playwright json report output
+   */
+  async processSpecs(specs: any) {
+    for (const spec of specs) {
+      if (spec.ok === true) {
+        continue
+      }
+
+      for (const test of spec.tests) {
+        for (const result of test.results) {
+          const expected = result.attachments[0]
+          const actual = result.attachments[1]
+          const diff = result.attachments[2]
+
+          this.copyActualImageToUpdatedSnapshotsFolder(actual)
+          await this.createCompositeDiffAndCopyToDiffOutputFolder(
+            expected,
+            actual,
+            diff,
+          )
+        }
+      }
+    }
+  }
+
+  /**
+   * Copy the image that was made by the tests to the updated_snapshots folder
+   * @param actual The {@link Attachment} info for the actual image that was generated by the test
+   */
+  copyActualImageToUpdatedSnapshotsFolder(actual: Attachment) {
+    // Copy actual file to updated_snapshots
+    const updatedSnapshotFile = path.join(
+      './updated_snapshots',
+      actual.name.replace('-actual.png', '-received.png'),
+    )
+    fs.mkdirSync(path.dirname(updatedSnapshotFile), {recursive: true})
+    fs.copyFileSync(actual.path, updatedSnapshotFile)
+  }
+
+  /**
+   * Create a composite image of the expected, diff, and actual images and copy it to the diff_output folder
+   * @param expected The {@link Attachment} info for the expect image that exists in the images_snapshot folder
+   * @param actual The {@link Attachment} info for the actual image that was generated by the test
+   * @param diff The {@link Attachment} info for the diff image that was generated by the test
+   */
+  async createCompositeDiffAndCopyToDiffOutputFolder(
+    expected: Attachment,
+    actual: Attachment,
+    diff: Attachment,
+  ) {
+    // Determine the max height based on the three images
+    const expectedMetadata = await sharp(expected.path).metadata()
+    const diffMetadata = await sharp(diff.path).metadata()
+    const actualMetadata = await sharp(actual.path).metadata()
+    const canvasHeight = Math.max(
+      expectedMetadata.height ?? 0,
+      diffMetadata.height ?? 0,
+      actualMetadata.height ?? 0,
+    )
+
+    // Determine the left position for the second and third images
+    const diffFileLeft = expectedMetadata.width ?? 0
+    const actualFileLeft = diffFileLeft + (diffMetadata.width ?? 0)
+
+    // Make the new file path
+    // Path format is: ./diff_output/test-file-name/image-name.png
+    const stitchedImagePath = path.join('./diff_output', diff.name)
+
+    // Make sure the parent directory structure exists
+    fs.mkdirSync(path.dirname(stitchedImagePath), {recursive: true})
+
+    // Stitch images together
+    const imageWidth = 1280
+
+    sharp({
+      create: {
+        width: imageWidth * 3,
+        height: canvasHeight,
+        channels: 4, // RGBA
```

Comment:
nitpick: delete or move "RGBA" comment?

Reply:
I'm going to keep it local to the line. It's there to make it quick to know what 4 stands for and moving it will break that association and require more context.

Reply:
Ohhh I thought it was associated with the line below. Sounds good!